### PR TITLE
Task/229 Stub api in trading e2e

### LINKED ISSUE #239: Convert tests to use stubbed api

### PR DESCRIPTION
**Pull request by mattrussell36**
- Split out vega wallet into own support class
- Add markets-page.feature (for market list page) and portfolio-page.feature
- Update market-order.feature to be deal-ticket.feature
- Add cy.intercepts for queries, with fixtures

Closes #239

### CODE REVIEW

File: apps/trading-e2e/src/fixtures/markets/active/chart.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (dexturr) [inline apps/trading-e2e/src/fixtures/markets/active/chart.json:1]: comment (non-blocking): not TS or anything so if the API changes and these files go out of date we won't be notified.

File: apps/trading-e2e/src/support/index.ts [hunk 2]
```diff
@@ -15,3 +15,12 @@
 
 // Import commands.js using ES2015 syntax:
 import '@vegaprotocol/cypress';
+
+// Utility to match GraphQL mutation based on the operation name
+export const hasOperationName = (req, operationName) => {
+  const { body } = req;
+  return (
+    // eslint-disable-next-line no-prototype-builtins
+    body.hasOwnProperty('operationName') && body.operationName === operationName
```
Comment (dexturr) [inline apps/trading-e2e/src/support/index.ts:24]: question (non-blocking): can we avoid this eslint disable?

```
'operationName' in body && ....
```

File: apps/trading-e2e/src/support/trading-windows/deal-ticket.js [hunk 3]
```diff
@@ -65,17 +63,21 @@ export default class DealTicketPage extends BasePage {
   }
 
   clickPlaceOrder() {
-    if (Cypress.env('bypassPlacingOrders' != true)) {
-      cy.getByTestId(this.placeOrderBtn).click();
-    }
+    // if (Cypress.env('bypassPlacingOrders' != true)) {
```
Comment (dexturr) [inline apps/trading-e2e/src/support/trading-windows/deal-ticket.js:66]: chore (blocking): commented out code

File: apps/trading-e2e/src/support/step_definitions/vega-wallet.step.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        signature: {
+          // sig value needs to be 'real' so sigToId function doesn't error out
+          value:
+            'd86138bba739bbc1069b3dc975d20b3a1517c2b9bdd401c70eeb1a0ecbc502ec268cf3129824841178b8b506b0b7d650c76644dbd96f524a6cb2158fb7121800',
+          algo: 'vega/ed25519',
+          version: 1,
+        },
+        From: {
+          PubKey:
+            // TODO: Use same pubkey as the test wallet
```
Comment (dexturr) [inline apps/trading-e2e/src/support/step_definitions/vega-wallet.step.js:22]: chore (non-blocking): todo? Issue, resolve or remove please :pray:

Comment (dexturr) [reply apps/trading-e2e/src/support/step_definitions/vega-wallet.step.js:22]: Possibly the best solution is to move this to the env file so people can chnage it for local ect.

File: apps/trading-e2e/src/support/step_definitions/deal-ticket.step.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  dealTicket.clickPlaceOrder();
+});
+
+When('I place a buy {string} market order with amount of 0', (orderType) => {
+  dealTicket.placeMarketOrder(true, 0, orderType);
+  dealTicket.clickPlaceOrder();
+});
+
+Then('order request is sent', () => {
+  // if (Cypress.env('bypassPlacingOrders' != true)) {
```
Comment (dexturr) [inline apps/trading-e2e/src/support/step_definitions/deal-ticket.step.js:32]: ?

**Review by dexturr — APPROVED** [review]
LGTM, few minor formatting things. I added some comments just o be explicit about the trade offs I saw us making here. They aren't blocking or need any action just merely to say "We know this and we are ok with it" 👍

File: apps/trading-e2e/src/fixtures/markets/active/chart.json (hunk 1 above)
Comment (mattrussell36) [reply apps/trading-e2e/src/fixtures/markets/active/chart.json]: converted to generator functions colocated with queries

**Review by jtsang586 — APPROVED** [review]

**Pull request merged**